shuup/utils/money.py

```diff
@@ -60,3 +63,30 @@ def unit_matches_with(self, other):
 
     def new(self, value):
         return type(self)(value, currency=self.currency)
+
+    def as_rounded(self, digits=None):
+        """
+        Returns the value rounded to the given `digits` if specified,
+        or to the fetched `decimal_places` from the Currency
+        loaded from db using the `currency` attribute of this instance.
+
+        In case of non-existent currency in the db,
+        the fallback `digits` will always be 0.
+
+        :param digits int:
+          The number of digits to round
+
+        :return A new value rounded to the given digits
+        """
+
+        if digits is None:
+            # the digits info is not cached yet, fetch it
+            if self.currency not in digits_cache:
+                from shuup.core.models._currencies import Currency
+
+                currency = Currency.objects.filter(code=self.currency).first()
+                digits_cache[self.currency] = currency.decimal_places if currency else 0
+
+            return self.new(numbers.bankers_round(self.value, digits_cache[self.currency]))
+        else:
+            return self.new(numbers.bankers_round(self.value, digits))
```

Comment:
@chessbr One of the requirements is to have merchant to be able to set the precision for currency. So if we have `DefaultPrecisionProvider` it must offer these in way more dynamic manner:
```python
# map of precisions for currencies
    currency_precisions = digits_to_precisions({
        'USD': 2,
        'EUR': 2,
        # fill this with more..
    })
```
 
What would be the use case of changing the `PrecisionProvider`? There are (imo) two ways of setting the precisions:
* Hard coded list of known currency precisions
* Merchant can override it

Like said before, `DefaultPrecisionProvider` should handle both cases. I like the idea of the `Provider` but can you guys think any real world example of why you would like to use other `Provider` instead of the `DefaultPrecisionProvider`? I can't think any now.